packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/CanonicalTypeProvider.cs

```diff
@@ -192,27 +191,39 @@ protected override FieldProvider[] BuildFields()
                         customField.Type.IsPublic,
                         customField.Type.IsStruct,
                         customField.Type.BaseType,
-                        TypeFactory.CreatePrimitiveCSharpTypeCore(inputEnumType!.ValueType));
+                        TypeFactory.CreatePrimitiveCSharpTypeCore(specType!));
                 }
             }
 
             return [..generatedFields, ..customFields];
         }
 
-        private static bool IsExtensibleEnum(InputModelProperty? inputProperty, out InputEnumType? inputEnumType)
+        private static bool IsCustomizedEnumProperty(InputModelProperty? inputProperty, CSharpType customType, out InputType? specValueType)
```

Comment:
```suggestion
        private static bool IsCustomizedEnumProperty(InputModelProperty? inputProperty, CSharpType customType, [NotNullWhen(true)] out InputType? specValueType)
```
thoughts? That would avoid having to use the null forgiving operator when we know the output shouldn't be null.